### PR TITLE
fix: ensure stable task fingerprints by sorting environment variables

### LINKED ISSUE #47: Unstable command envs sort cause cache miss
First run

```bash
vp lint                                                                                                          node(v22.17.0) 
Cache Not Found, executing task
> NODE_PACKAGE_MANAGER=vite-plus JS_RUNTIME_VERSION=22.18.0 JS_RUNTIME_NAME=node '/Users/fengmk2/.volta/tools/image/packages/@voidzero-dev/global/lib/node_modules/@voidzero-dev/global/node_modules/oxlint/bin/oxlint'
```

Again

```bash
vp lint                                                                                                          node(v22.17.0) 
Cache Not Found, executing task
> JS_RUNTIME_NAME=node NODE_PACKAGE_MANAGER=vite-plus JS_RUNTIME_VERSION=22.18.0 '/Users/fengmk2/.volta/tools/image/packages/@voidzero-dev/global/lib/node_modules/@voidzero-dev/global/node_modules/oxlint/bin/oxlint'
```

The order of the command's environment variables changes, causing the cache to miss.

### PR DESCRIPTION
**Pull request by fengmk2**
## Summary
- Fixed unstable task fingerprints caused by non-deterministic HashMap/HashSet iteration order
- Changed `TaskParsedCommand.envs` from HashMap to BTreeMap for stable iteration
- Added sorting when populating `envs_without_pass_through` from HashSet

## Problem
Task fingerprints were unstable across runs due to HashMap/HashSet's non-deterministic iteration order, causing unnecessary cache misses.

## Solution
1. **Use BTreeMap for parsed command envs**: Ensures environment variables are always iterated in alphabetical order
2. **Sort env names before processing**: When building `envs_without_pass_through`, sort the HashSet keys first
3. **Added comprehensive tests**: 7 new unit tests validate fingerprint stability across multiple scenarios

## Test Coverage
- `test_task_parsed_command_stable_env_ordering`: Verifies BTreeMap maintains alphabetical order
- `test_task_parsed_command_serialization_stability`: Ensures binary serialization is deterministic
- `test_task_envs_stable_ordering`: Tests sorted env processing produces consistent results
- `test_command_fingerprint_stable_with_multiple_envs`: Validates overall fingerprint stability
- Plus 3 additional tests for various edge cases

Closes #47

Co-authored-by: Claude <noreply@anthropic.com>

### CODE REVIEW

File: crates/vite_task/src/execute.rs [hunk 1]
```diff
@@ -221,7 +221,10 @@ impl TaskEnvs {
             .collect();
 
         let mut envs_without_pass_through = HashMap::<Str, Str>::new();
-        for name in &task.config.envs {
+        // Sort env names to ensure stable fingerprint
+        let mut sorted_env_names: Vec<_> = task.config.envs.iter().collect();
+        sorted_env_names.sort();
+        for name in sorted_env_names {
```
Comment (wan9chi) [inline crates/vite_task/src/execute.rs]: What are these changes for? Seems unnecessary?

Comment (fengmk2) [reply crates/vite_task/src/execute.rs]: fixed
